### ISSUE TITLE
Use TorchMetrics for dataset-wide metric tracking

### ISSUE THREAD

**Issue by PierreBoyeau** [scverse/scvi-tools#1225]
The current [codebase](https://github.com/YosefLab/scvi-tools/blob/dca3f8f7f3645e60e7e1f14168a787785fd1a32d/scvi/train/_trainingplans.py#L161) recomputes the elbo for the full training data after each epoch. While discussions with @adamgayoso  suggest that this step does not constitute a major bottleneck, this step is not necessary. 

TorchMetrics provides a cleaner way to aggregate batch-level metrics, and should replace the current method.

**Comment by watiss** [comment]
@PierreBoyeau FYI we already have an issue related to the same: https://github.com/YosefLab/scvi-tools/issues/1113. Might want to keep one open and close the other as a dupe.
